RyuSocks/Auth/UsernameAndPassword.cs

```diff
@@ -9,21 +9,23 @@ namespace RyuSocks.Auth
     [AuthMethodImpl(0x02)]
     public class UsernameAndPassword : IProxyAuth
     {
-        public int WrapperLength { get; }
+        // Set to 0, Since there is no wrapping happening here
+        public int WrapperLength => 0;
 
         public void Authenticate()
         {
+            // Either requires a database or an authentication method given by the server 
             throw new NotImplementedException();
```

Comment:
The authentication method is Username/Password authentication. This method was supposed to be called from the client, so it needs to implement all the steps necessary to complete the authentication.

However this interface method is also missing a parameter which allows it to access the socket, so it can actually  communicate with the server. If you want you could adjust the interface for that and maybe also add a new interface method for the authentication sub-negotiation on the server side.

Reply:
I just pushed a commit 44d71ac to address this by adjusting the signature of this method.